larray/core/array.py

```diff
@@ -2942,12 +2942,15 @@ def with_total(self, *args, **kwargs):
 
         Parameters
         ----------
-        args
-        kwargs
-        op : aggregate function
+        *args : int or str or Axis or Group or any combination of those, optional
+            Axes or groups along which to compute the aggregates. Passed groups should be named. 
+            Defaults to aggregate over the whole array.
+        op : aggregate function, optional 
             Defaults to `sum()`.
-        label : scalar value
-            label to use for the total. Defaults to "total".
+        label : scalar value, optional 
+            label to use for the total. Applies only to aggregated axes, not groups. Defaults to "total".
+        **kwargs : Axis or Group, optional
```

Comment:
Sorry, to be picky, but this is wrong again. kwargs can also be scalars or combinations of groups/scalars and cannot be Axis.

Reply:
I am often confused about arguments of methods like getitem and _aggregate. If one day we only support Python 3, it would be great to use type hints in method signatures (you already proposed that but it's tedious to have something working on both Python 2 and 3).

Reply:
I don't see how type hints would help your confusion. That's another syntax for the same thing. The reason it could be interesting is that tools exist to check the args values actually used when called the function respect the type mentioned in the type hints, whereas no such AFAIK for numpy docstrings (but we could implement one ;-))

Reply:
kwargs can be scalars and groups but also string or not?

Reply:
You raise an interesting point. Whenever I speak about scalars I always use the numpy definition of scalar, which includes strings. I admit this is not obvious... Unsure what we should do about this. Avoid the term or???